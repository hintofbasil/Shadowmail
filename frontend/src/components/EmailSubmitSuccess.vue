<template>
  <div v-if="email">
    <div class="new-email-success-response">
      <i class="material-icons descriptor-icon descriptor-icon-mail">
        mail_outline
      </i>
      <br />
    </div>
    <div class="input-then-button-layout">
      <button
        type="submit"
        value="content_copy"
        title="Copy to clipboard"
        id="email-copy-button"
        v-on:click="copy"
      >
        <i class="material-icons descriptor-icon submit-material-icon">
          content_copy
        </i>
        <br />
      </button>
      <span class="submit-span">
        <input
          type="email"
          id="new-email-success-text"
          v-model="email"
          readOnly
        />
      </span>
      <div id="copy-status">
        {{ copyStatus }}
      </div>
    </div>
  </div>
</template>

<script>
export default {
  data: () => {
    return {
      copyStatus: null
    };
  },
  methods: {
    copy: function(event) {
      event.preventDefault();
      const emailInput = document.getElementById("new-email-success-text");
      emailInput.select();
      const success = document.execCommand("copy");
      this.copyStatus = success ? "Copied!" : "Unable to copy";
    }
  },
  props: ["email"]
};
</script>

<style lang="scss" scoped>
@import "../colors";

.input-then-button-layout {
  margin-bottom: 2rem;

  > :first-child {
    float: right;
    padding: 0;
    width: 38px; // Height is set in Skeleton
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }

  span {
    display: block;
    overflow: hidden;

    > :first-child {
      width: 100%;
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
      border-right: none;
    }
  }
}

.new-email-success-response {
  text-align: center;
  color: $success-text-color;
}

.submit-material-icon {
  font-size: 20px;
  line-height: 38px;
}

#new-email-success {
  margin-bottom: 5%;
}

#new-email-error {
  margin-bottom: 5%;
}

#copy-status {
  text-align: center;
}

.descriptor-icon-mail {
  width: 50%;
  font-size: 4em;
}
</style>
