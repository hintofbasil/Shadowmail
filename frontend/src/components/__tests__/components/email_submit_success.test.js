import React from 'react';
import renderer from 'react-test-renderer';

import EmailSubmitSuccess from '../../email_submit_success';

describe('render', () => {
  it('renders successfully', () => {
    const tree = renderer
      .create(
        <EmailSubmitSuccess
          email="test@example.com"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
