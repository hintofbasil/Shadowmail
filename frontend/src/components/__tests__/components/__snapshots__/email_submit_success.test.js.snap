// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders successfully 1`] = `
Array [
  <div
    className="new-email-success-response"
  >
    <i
      className="material-icons descriptor-icon"
    >
      mail_outline
    </i>
    <br />
  </div>,
  <div
    className="input-then-button-layout"
  >
    <button
      id="email-copy-button"
      onClick={[Function]}
      title="Copy to clipboard"
      type="submit"
      value="content_copy"
    >
      <i
        className="material-icons descriptor-icon submit-material-icon"
      >
        content_copy
      </i>
      <br />
    </button>
    <span
      className="submit-span"
    >
      <input
        id="new-email-success-text"
        readOnly={true}
        type="email"
        value="test@example.com"
      />
    </span>
    <div
      id="copy-status"
    >
      
    </div>
  </div>,
]
`;
